command/pipe.go

```diff
@@ -0,0 +1,354 @@
+	}
+}
+
+func (c Pipe) doUpload(ctx context.Context, dsturl *url.URL) error {
+	err := c.shouldOverride(ctx, dsturl)
+	if err != nil {
+		if errorpkg.IsWarning(err) {
+			printDebug(c.op, err, nil, dsturl)
+			return nil
+		}
+		return err
+	}
+
+	// override destination region if set
+	if c.dstRegion != "" {
+		c.storageOpts.SetRegion(c.dstRegion)
+	}
+	dstClient, err := storage.NewRemoteClient(ctx, dsturl, c.storageOpts)
+	if err != nil {
+		return err
+	}
+
+	metadata := storage.NewMetadata().
+		SetStorageClass(string(c.storageClass)).
+		SetSSE(c.encryptionMethod).
+		SetSSEKeyID(c.encryptionKeyID).
+		SetACL(c.acl).
+		SetCacheControl(c.cacheControl).
+		SetExpires(c.expires)
+
+	if c.contentType != "" {
+		metadata.SetContentType(c.contentType)
+	} else {
+		metadata.SetContentType(guessStdinContentType(dsturl))
+	}
+
+	if c.contentEncoding != "" {
+		metadata.SetContentEncoding(c.contentEncoding)
+	}
+
+	err = dstClient.Put(ctx, &stdin{file: os.Stdin}, dsturl, metadata, c.concurrency, c.partSize)
+
+	if err != nil {
+		return err
+	}
+
+	msg := log.InfoMessage{
+		Operation:   c.op,
+		Source:      nil,
+		Destination: dsturl,
+		Object: &storage.Object{
+			StorageClass: c.storageClass,
+		},
+	}
+	log.Info(msg)
+
+	return nil
+}
+
+// shouldOverride function checks if the destination should be overridden if
+// the destination object and given pipe flags conform to the
+// override criteria.
+func (c Pipe) shouldOverride(ctx context.Context, dsturl *url.URL) error {
+	// if not asked to override, ignore.
+	if !c.noClobber {
+		return nil
+	}
+
+	dstClient, err := storage.NewClient(ctx, dsturl, c.storageOpts)
+	if err != nil {
+		return err
+	}
+
+	dstObj, err := getObject(ctx, dsturl, dstClient)
+	if err != nil {
+		return err
+	}
+
+	// if destination not exists, no conditions apply.
+	if dstObj == nil {
+		return nil
+	}
+
+	var stickyErr error
+	if c.noClobber {
+		stickyErr = errorpkg.ErrObjectExists
+	}
+
+	return stickyErr
+}
+
+func validatePipeCommand(c *cli.Context) error {
+	if c.Args().Len() != 1 {
+		return fmt.Errorf("expected destination argument")
+	}
+
+	dst := c.Args().Get(0)
+
+	dsturl, err := url.New(dst, url.WithRaw(c.Bool("raw")))
+	if err != nil {
+		return err
+	}
+
+	// destination must be a file
+	if dsturl.IsBucket() || dsturl.IsPrefix() {
+		return fmt.Errorf("target %q must be a file", dsturl)
+	}
+
+	// wildcard destination can not be used with pipe
+	if dsturl.IsWildcard() {
+		return fmt.Errorf("target %q can not contain glob characters", dst)
+	}
+
+	return nil
+}
+
+func guessStdinContentType(dsturl *url.URL) string {
+	contentType := mime.TypeByExtension(filepath.Ext(dsturl.Absolute()))
+	if contentType == "" {
+		return "application/octet-stream"
+	}
+	return contentType
+}
+
+// if we pass os.Stdin directly to Put function AWS SDK throws an error
+// ReadRequestBody: unable to initialize upload caused by: seek /dev/stdin: illegal seek
+// due to that stdin struct has only file without seek
```

Comment:
It would be good to follow the convention; 
https://tip.golang.org/doc/comment#type

Instead of dumping error how about something more abstract like;
```
// stin is an io.Reader adapter for os.File, enabling it to function solely as
// an io.Reader. The AWS SDK, which accepts an io.Reader for multipart uploads,
// will attempt to use io.Seek if the reader supports it. However, os.Stdin is
// a specific type of file that can not seekable.

```

Reply:
Thanks for the tip. I will update the comment.